src/ansible_compat/_compat.py

```diff
@@ -0,0 +1,31 @@
+"""Internal glue."""
+import logging
+import sys
+from typing import Callable, TypeVar
+
+try:
+    import tenacity
+except ImportError:
+    pass
+
+WrappedFn = TypeVar("WrappedFn", bound=Callable)
+_logger = logging.getLogger(__name__)
+
+
+def noop_dec(f: WrappedFn) -> WrappedFn:
+    """Decorate function with a no-op."""
+    return f
+
+
+if "tenacity" in sys.modules:
```

Comment:
Plz set a var during the import attempt and use it then. It's better than doing a list scan in the import-time (just like `'tenacity' in globals()`)

Reply:
I did not had a problem with original approach but the linters did, especially mypy which was upset about tenacity being either None or a ModuleType, feel free to push to the branch with a fix. As far is passing I don't mind. I will stop doing forced pushes in order to be able to collab on it.

Reply:
Okay, then just add an explanation comment for anyone having a bright idea to refactor this in the future.